hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/action/ActionDefinition.java

```diff
@@ -0,0 +1,132 @@
+ * limitations under the License.
+ */
+package org.hawkular.alerts.api.model.action;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+/**
+ * An action is the abstract concept of ta consequence of an event.
+ *
+ * Actions are processed by plugins, and plugins offer a map of properties to personalize an action.
+ * An ActionDefinition stores which properties will be used for a specific action in a specific plugin.
+ *
+ * A Trigger definition can be linked with a list of action definitions.
+ *
+ * Alert engine will instantiate a specific Action based on its ActionDefinition.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+public class ActionDefinition implements Serializable {
+
+    @JsonInclude
+    private String tenantId;
+
+    @JsonInclude
+    private String actionPlugin;
+
+    @JsonInclude
+    private String actionId;
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    private Map<String, String> properties;
+
+    public ActionDefinition() {
+    }
+
+    public ActionDefinition(String tenantId, String actionPlugin, String actionId) {
+        this(tenantId, actionPlugin, actionId, Collections.EMPTY_MAP);
+    }
+
+    public ActionDefinition(String tenantId, String actionPlugin, String actionId,
+                            Map<String, String> properties) {
+        this.tenantId = tenantId;
+        this.actionPlugin = actionPlugin;
+        this.actionId = actionId;
+        this.properties = new HashMap<>(properties);
+    }
+
+    public String getTenantId() {
+        return tenantId;
+    }
+
+    public void setTenantId(String tenantId) {
+        this.tenantId = tenantId;
+    }
+
+    public String getActionPlugin() {
+        return actionPlugin;
+    }
+
+    public void setActionPlugin(String actionPlugin) {
+        this.actionPlugin = actionPlugin;
+    }
+
+    public String getActionId() {
+        return actionId;
+    }
+
+    public void setActionId(String actionId) {
+        this.actionId = actionId;
+    }
+
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    public void setProperties(Map<String, String> properties) {
+        this.properties = properties;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        ActionDefinition that = (ActionDefinition) o;
+
+        if (tenantId != null ? !tenantId.equals(that.tenantId) : that.tenantId != null) return false;
+        if (actionPlugin != null ? !actionPlugin.equals(that.actionPlugin) : that.actionPlugin != null) return false;
+        if (actionId != null ? !actionId.equals(that.actionId) : that.actionId != null) return false;
+        return properties != null ? properties.equals(that.properties) : that.properties == null;
```

Comment:
Point is that properties are mutable, so yes, I think those properties are needed on equals/hashCode